packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/point_in_time_finder.ts

```diff
@@ -93,7 +93,12 @@ export class PointInTimeFinder<T = unknown, A = unknown>
         await this.close();
       }
 
-      yield results;
+      // do not yield first page if empty, unless there are aggregations
+      // (in which case we always want to return at least one page)
+      if (lastResultsCount > 0 || this.#findOptions.aggs) {
+        yield results;
+      }
```

Comment:
Even if this is an edge case, it is possible to specify aggregations with a PIT search, and this is effectively done at least once in our code: 

https://github.com/elastic/kibana/blob/0ae3b23752157a9c8c6dc578b4a6fbc8c26f89d6/x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts#L266-L276

Note that this specific usage could be adapted (given there isn't any reason to use a PIT search here), but we scenario remains valid more globally, so we still have to check for aggregations and yield, otherwise there's no way for the API consumers to retrieve their aggregation results in case of empty hits.

Reply:
cc @elastic/fleet as you may want to adapt this snippet to use a standard `SOR.find` call instead of a PIT given you break on first page.